src/pages/add_new_contact_page.py

```diff
@@ -70,90 +64,74 @@ def cancel_from_add_new_contact_page(self):
 
         logger.info("Cancel from add new contact page")
 
-        WebDriverWait(self.browser, 10).until(
-            EC.visibility_of_element_located(
-                AddNewContactPageLocators.CANCEL_BUTTON
-            )
-        )
-
-        cancel_button = self.browser.find_element(
-            *AddNewContactPageLocators.CANCEL_BUTTON
+        cancel_button = self.wait_for_element_ready(
```

Comment:
Better an easy way is to just cover it by 1 function:
```suggestion
        cancel_button = self.click_button(*AddNewContactPageLocators.CANCEL_BUTTON)
```

Define this method in the Base class:
```
def click_button(locator):
        self.find_element(*locator)
        self.click()

def click_button(locator):
        # Find and wait for any element

def click(el):
        # click on any element
```

As a result, you will use 1 function to search and click on any element.
Review your repository, there are a lot of places with this kind of mistake. It also relates to other types of actions: send a text to the input field and diff. search conditions